src/main/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAreasOfLawService.java

```diff
@@ -2,34 +2,77 @@
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.entity.CourtAreaOfLaw;
 import uk.gov.hmcts.dts.fact.exception.NotFoundException;
 import uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw;
+import uk.gov.hmcts.dts.fact.repositories.CourtAreaOfLawRepository;
 import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
 
-import java.util.List;
+import java.util.*;
+import java.util.stream.Collectors;
 
 import static java.util.stream.Collectors.toList;
 
 @Service
 public class AdminCourtAreasOfLawService {
 
     private final CourtRepository courtRepository;
+    private final CourtAreaOfLawRepository courtAreaOfLawRepository;
 
     @Autowired
-    public AdminCourtAreasOfLawService(final CourtRepository courtRepository) {
+    public AdminCourtAreasOfLawService(final CourtRepository courtRepository, final CourtAreaOfLawRepository courtAreaOfLawRepository) {
         this.courtRepository = courtRepository;
+        this.courtAreaOfLawRepository = courtAreaOfLawRepository;
     }
 
     public List<AreaOfLaw> getCourtAreasOfLawBySlug(final String slug) {
-        return courtRepository.findBySlug(slug)
-            .map(c -> c.getAreasOfLaw()
-                .stream()
-                .map(AreaOfLaw::new)
-                .collect(toList()))
+
+        Court court = courtRepository.findBySlug(slug).orElseThrow(() -> new NotFoundException(slug));
+        List<CourtAreaOfLaw> courtAreaOfLawList = courtAreaOfLawRepository.getCourtAreaOfLawByCourtId(court.getId());
```

Comment:
Very small thing. I think the variable courtAreaOfLawList is not dong much here. It is assigned and straight away get used in the next line. Is the not cleaner to combine these 2 in one like below?

return courtAreaOfLawRepository.getCourtAreaOfLawByCourtId(court.getId())
            .stream()
            .map(aol -> new AreaOfLaw(aol.getAreaOfLaw(), aol.getSinglePointOfEntry()))
           .collect(toList());